http/middleware.go

```diff
@@ -84,20 +87,21 @@ func WithHeaderAuthorization(opts ...AuthorizationOption) func(http.Handler) htt
 // Tries a cached value first, but if there's no value or the entry
 // has expired, it will fetch the JWK set from the API and cache the
 // value.
-func getJWK(ctx context.Context, jwksClient *jwks.Client, kid string) (*clerk.JSONWebKey, error) {
+func getJWK(ctx context.Context, jwksClient *jwks.Client, kid string, clock clerk.Clock) (*clerk.JSONWebKey, error) {
 	if kid == "" {
 		return nil, fmt.Errorf("missing jwt kid header claim")
 	}
 
 	jwk := getCache().Get(kid)
-	if jwk == nil {
+	if jwk == nil || !getCache().IsValid(kid, clock.Now().UTC()) {
```

Comment:
💡 Not sure if that's difficult but perhaps the clock should be a field of `jwkCache `? That would solve the issue of checking for expiration inside `Get` if we want to retain this approach.

Actually, maybe ignore that I see that `getCache` is a singleton.